sodym/named_dim_arrays.py

```diff
@@ -385,42 +379,48 @@ def check_id0(self):
 
 
 class Flow(NamedDimArray):
-    """The values of Flow objects are the main computed outcome of the MFA system. A flow connects two processes. Its
-    name is set as a combination of the names of the two processes it connects.
+    """The values of Flow objects are the main computed outcome of the MFA system.
+    A Flow object connects two :py:class:`Process` objects.
+    The name of the Flow object is set as a combination of the names of the two processes it connects.
 
-    Note that it is a subclass of NamedDimArray, so most of the methods are defined in the NamedDimArray class.
-    """
+    Flow is a subclass of :py:class:`NamedDimArray`, so most of its methods are inherited.
 
-    from_process: Optional[Process] = None
-    to_process: Optional[Process] = None
-    from_process_name: str
-    to_process_name: str
+    **Example**
+
+    >>> from sodym import DimensionSet, Flow, Process
+    >>> goods = Dimension(name='Good', letter='g', items=['Car', 'Bus', 'Bicycle'])
+    >>> time = Dimension(name='Time', letter='t', items=[1990, 2000, 2010, 2020, 2030])
+    >>> dimensions = DimensionSet([goods, time])
+    >>> fabrication = Process(name='fabrication', id=2)
+    >>> use = Process(name='use', id=3)
+    >>> flow = Flow(from_process='fabrication', to_process='use', dims=dimensions)
+
+    In the above example, we did not pass any values when initialising the Flow instance,
+    and these would get filled with zeros.
+    See the validation (filling) method in :py:class:`NamedDimArray`.
+    """
+    model_config = ConfigDict(protected_namespaces=())
 
-    # TODO: Define flow by passing process objects, rather than from flow definition.
+    from_process: Process
+    to_process: Process
+    from_process_name: Optional[str] = None
+    to_process_name: Optional[str] = None
 
-    @model_validator(mode="before")
-    def check_process_names(v):
-        if "from_process" in v:
-            if v["from_process"].name != v["from_process_name"]:
-                raise ValueError("Missmatching process names in Flow object")
-        if "to_process" in v:
-            if not v["to_process"].name != v["to_process_name"]:
-                raise ValueError("Missmatching process names in Flow object")
-        return v
+    @model_validator(mode="after")
+    def check_process_names(self):
+        if self.from_process_name and self.from_process.name != self.from_process_name:
+            raise ValueError("Missmatching process names in Flow object")
+        self.from_process_name = self.from_process.name
+        if self.to_process_name and self.to_process.name != self.to_process_name:
+            raise ValueError("Missmatching process names in Flow object")
+        self.to_process_name = self.to_process.name
+        return self
 
     @model_validator(mode="after")
     def flow_name_related_to_proccesses(self):
         self.name = f"{self.from_process_name} => {self.to_process_name}"
         return self
```

Comment:
This could get a check that name is currently "unnamed", i.e. that we are not overwriting a custom name. If a custom name is present, we could just leave it and not overwrite it.

Reply:
Deleted this